### PR TITLE
feat(deploy): added kubernetes config files

### PR DESCRIPTION
**Pull request by alvarobernal2412**
This pull request includes significant updates to the Docker and Kubernetes configurations, as well as minor improvements to the package and API documentation. The most important changes are listed below:

### Docker and Kubernetes Configuration:

* [`docker-compose.yml`](diffhunk://#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L23-L32): Added new environment variables for MongoDB, RabbitMQ, and JWT configuration, and removed the volume mount.
* [`kubernetes/appointment-deployment.yaml`](diffhunk://#diff-8880e64d1c0f9a1525e8d0850b43fe9cee195ce802397a45765296fac51a4392R1-R81): Created a new deployment configuration for the `appointment` service, including environment variables and container specifications.
* [`kubernetes/mongodb-deployment.yaml`](diffhunk://#diff-37303c2921f0256f198ee864e1b2ca069f8e3485b6a6513a627c5c3e870f0d00R1-R49): Added a deployment configuration for MongoDB with environment variables and volume mounts.
* [`kubernetes/rabbitmq-deployment.yaml`](diffhunk://#diff-8cd155b799d28ced58416ffc464f0f19c14f2be611bd5c046ccc22e7fb22aee8R1-R40): Added a deployment configuration for RabbitMQ with environment variables.

### Service Definitions:

* [`kubernetes/appointment-service.yaml`](diffhunk://#diff-128799beff3e28ec5469da66f76b925f64c778e742c738216dd712fd353d4257R1-R18): Created a new service configuration for the `appointment` service with a LoadBalancer type.
* [`kubernetes/mongodb-service.yaml`](diffhunk://#diff-4094af2a5f5058f456d8d49437f645bf8f1abef40bcb804acc4425cd742a9600R1-R18): Added a service configuration for MongoDB with a ClusterIP type.
* [`kubernetes/rabbitmq-service.yaml`](diffhunk://#diff-3396b23e2843d16b17ba179eb5c8a9f9cc68b0fe78444a8ddc17976cf71ef07dR1-R18): Added a service configuration for RabbitMQ with a LoadBalancer type.

### Persistent Volume:

* [`kubernetes/mongo-data-appointments-persistentvolumeclaim.yaml`](diffhunk://#diff-dbc16b03d4f29611d239c75e612116c31afeaa1ea401e0d82ad854b08b41bdc1R1-R12): Created a PersistentVolumeClaim for MongoDB data storage.

### Other Changes:

* [`openapi.yaml`](diffhunk://#diff-d910ba2ef878f7db0223a966b81c8b3f3b65027bb39e4431bb05140171eece39R12-R13): Added a new server entry for the production environment.
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L14-R14): Updated the `start` script to remove the `--env-file` option.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -20,16 +20,25 @@ services:
...
+      - MONGOURL=${MONGOURL}
+      - NODE_ENV=production
+      - RABBIT_USERNAME=${RABBIT_USERNAME}
+      - RABBIT_PASSWORD=${RABBIT_PASSWORD}
+      - RABBIT_HOST=${RABBIT_HOST}
+      - RABBIT_PORT=${RABBIT_PORT}
+      - RABBIT_VHOST=${RABBIT_VHOST}
+      - API_PREFIX=/api/v1
+      - JWT_SECRET=${JWT_SECRET}
+      - KAFKA_HOST=${KAFKA_HOST}
```
Comment (adrrf) [inline docker-compose.yml:35]: need to add dragonfly service

**Review by adrrf — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (adrrf) [reply docker-compose.yml:35]: also add the readme from template in this repo

Comment (alvarobernal2412) [reply docker-compose.yml:46]: Resolved in the commit `62348a81a7e974d7714c4058255e5ae8da78d204`

**Review by adrrf — APPROVED** [review]
after 6 tries, Terminator Thumb we need to have a good documentation so everyone doesnt suffer as i did

**Pull request merged**